### ISSUE TITLE
bug(gemma4/cuda): decode collapses to <pad> after 1-3 tokens (single-query maskless attention + quantized GEMV); mlx_lm decodes coherently

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#688]
## Summary
On CUDA (GB10), gemma-4 (gemma-4-12b-it-4bit and the uniform-4bit requant) greedy decode collapses to `<pad>` (id 0) spam after only 1 to 3 generated tokens, even when the prompt is byte-identical to a reference that decodes correctly. This is a decode-path (single-query, KV-cache) forward bug specific to mlxcel: the prefill and first-token logits are correct, but the incremental decode diverges. Discovered while fixing the gemma-4 chat-template rendering (#686, PR #687); that fix made the first token correct but the decode collapse is a separate, still-open defect and is the remaining blocker to gemma-4 usability.

## Evidence (GB10, release, greedy temp-0)
On the exact same model and byte-identical chat prompt ("Explain in two sentences why the sky is blue."):
- Apple mlx_lm gemma4_text decodes 50 tokens coherently: "The sky appears blue because of a phenomenon called Rayleigh scattering, where Earth's atmosphere scatters shorter wavelengths of light more effectively. Because blue light travels in shorter, smaller waves, it is scattered in all directions by gas molecules more than colors".
- mlxcel gemma-4-12b decodes "The" then all `<pad>`.
- mlxcel gemma-3-4b (control) decodes coherently, so the decode path is fine for non-gemma-4 models.

The forward is NOT the cause of the first token: mlxcel fed the reference's exact rendered prompt generates the correct first token ("Paris" for a capital-of-France prompt), and mlxcel's teacher-forced prefill perplexity matches mlx_lm within 0.05 nats. So the bug is in the INCREMENTAL DECODE path (single-query attention over the growing KV cache), not prefill.

## Localization (already narrowed)
- `MLXCEL_DISABLE_SINGLE_QUERY_MASKLESS=1` lengthens the coherent run from ~2 to ~24 tokens (uniform-4bit), so the single-query maskless decode-attention path is the dominant contributor.
- The 8-bit MLP GEMV (the shipped mixed-quant gemma-4-12b) shortens the coherent run further (~24 to ~6 tokens vs the uniform-4bit ~24), so a quantized-GEMV precision interaction is a secondary contributor.
- Collapse onset at cache length ~25 is far below the sliding_window of 1024, so it is not sliding-window overflow.

## Suspects
The single-query decode attention for gemma-4's exotic layer mix: the global/full-attention layers (proportional RoPE, global_head_dim 512, num_global_key_value_heads 1, attention_k_eq_v true) and how their KV cache is read/rotated during single-token decode; the proportional-RoPE offset applied per decode step; the dual head_dim (512 global vs 256 sliding) KV layout during decode; and the maskless single-query attention assuming a uniform head_dim or causal-by-construction that does not hold for the global layers. Plus a quantized-GEMV precision path that accelerates the collapse. Compare the mlxcel decode attention dispatch against mlx_lm's gemma4_text decode (which is coherent) layer-type by layer-type.

## Reproduction
`mlxcel generate -m models/gemma-4-12b-it-4bit -p "Explain in two sentences why the sky is blue." -n 40 --temp 0` (collapses); add `MLXCEL_DISABLE_SINGLE_QUERY_MASKLESS=1` to see the longer partial run. Reference: the mlx_lm oracle in the investigation venv decodes coherently.

## Acceptance criteria
- [x] gemma-4-12b-it-4bit greedy decode stays coherent for at least 100 tokens on a normal chat prompt, matching the mlx_lm reference token trajectory closely (near-tie FP drift allowed)
- [x] The uniform-4bit and mixed 8-bit-MLP variants both decode coherently (or the quantized-GEMV factor is separately characterized and fixed)
- [x] gemma-3 and non-gemma decode unchanged (no regression)
- [x] Decode tok/s within a few percent of the current baseline
- [x] A test or documented reference-parity check pins the fix

## Related
- #686 / PR #687 (chat-template + tokenizer BOS fixes that make the first token correct; this decode bug is the remaining part)
- MLXCEL_DISABLE_SINGLE_QUERY_MASKLESS (docs/environment-variables.md), #675/#680/#683 (gemma-4 attention/decode perf work, all of which measured throughput not correctness)
